### DISCUSSION TITLE
Large Images

### DISCUSSION THREAD

**Discussion by fcuenca4** [esimov/pigo#66] [Q&A] [answered]
Hi! 

I'm currently trying to process a large dataset of images, different sizes and formats. 
In my tests I've found that some photos contain false positives - I'm aware that's entirely possible due to the nature of the Pixel Intensity Comparison-based Object detection algorithm.

Now, I'm trying to understand what is the reason of some of my images being extremely weird in terms of detections. 
This is the image I'm using to test detections:
 
![shutterstock_398308393](https://github.com/esimov/pigo/assets/9162072/3eafd5a7-6d21-4651-abfb-31223f0a1cb6)

and this is the outcome: 

![output](https://github.com/esimov/pigo/assets/9162072/aed16387-0fe6-4623-a74d-ced6099d4734)

As you can see there are some faces are correctly detected whilst some of them are in weird places. 
Do you think that's related to the noise of the photo? I've been reading this paper (https://arxiv.org/pdf/1305.4537.pdf) and it doesn't recommend running any denoise pre running face detection, does the same apply to your implementation of `pico`? 

I'm using `ioU threhold: 0.3` and 
```
	cp := facedetection.CascadeParams{
		MinSize:     50,
		MaxSize:     1000,
		ShiftFactor: 0.1,
		ScaleFactor: 1.1,
	}
```

My last question is about resizing images before running `pigo`, do you recommend? I've noticed that resizing an image to a reasonable size makes the analysis way faster without much confidence reduction.

Thanks for your help,
Fran

**Comment by esimov** [comment, marked as answer]
The reason why you are getting false positive detections is because you need to tweak a little bit on the cascade params, specifically on `MinSize` and `MaxSize`. These parameters defines the minimum and maximum detection window. In your case, because you are running the detector over high resolution images, you have to increase them, otherwise you are getting these false positives. Try with `-min=400` and `-max=2000`. These were giving me correct results.

**Reply by fcuenca4** [reply]
I'm resizing the image to 512x512 before running pigo so those values will produce no results. 
The reason why I'm resizing is because of speed, do you see any issues with that?

In this case these cascade params produce perfect results for this photo: 
```
	cp := facedetection.CascadeParams{
		MinSize:     80,
		MaxSize:     500,
		ShiftFactor: 0.1,
		ScaleFactor: 1.1,
	}

**Reply by esimov** [reply]
It's perfectly fine to resize large images before face detecrion. The idea is that you have to adjust the cascade params for large images. The max size has been reduced as default parameter  for performance reasons.
